cleverhans/model_abs.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta
+from keras.models import Model
+
+
+class ModelAbstraction:
+    """
+    An abstract interface for a model that exposes the required functionalities
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model):
+        """
+        :param model: A function that takes a symbolic input and returns the
+                      symbolic output for the model's predictions.
+        """
+        raise NotImplementedError('Model abstraction not implementated.')
+
+    def get_hidden(self, layer):
+        """
+        :return: A symbolic representation of the hidden features
+        :param layer: The name of the hidden layer to return features at.
```

Comment:
This requires that all classes that inherit `ModelAbstraction` keep track of the name of layers when defining the model. It would be nice to formalize this in the documentation of `__init__`